### PR TITLE
docs: add contribution guideline

### LINKED ISSUE #116: [Docs]: Contribution Guideline
Add a contribution guideline for this project when creating issues and pull requests.

### PR DESCRIPTION
**Pull request by Olanetsoft**
Fixes #116

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+If you are new to the Axelar Network, we encourage you to read the [Axelar Documentation](https://docs.axelar.network/) to learn more about the Axelar Network and how to get started. 
+
+If you are new to contributing to open-source projects, we encourage you to read [How to Contribute to Open Source](https://opensource.guide/how-to-contribute/). If you are new to contributing to Axelar, we encourage you to read [`CONTRIBUTING.md`](./CONTRIBUTING.md) in the Axelar examples repository.
+
+
+Before you start contributing, you need to:
+
+1. Fork the repository and clone it to your local machine.
+2. Create a new branch for your contribution.
```
Comment (StephenFluin) [inline CONTRIBUTING.md:26]: Is this a required step? Open source contributors won't have permission to push to our repo anyway, so forking effectively is a new branch.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+3. Install the dependencies for the project as indicated in the [README.md](https://github.com/axelarnetwork/axelar-examples#readme) file.
+4. Make your changes and commit them to your local repository.
+
+Once you've completed these steps, you can contribute to Axelar Examples!
```
Comment (StephenFluin) [inline CONTRIBUTING.md:30]: I'm a little confused on the ordering, isn't this the actual contribution itself, not a prerequisite?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+If you are new to contributing to open-source projects, we encourage you to read [How to Contribute to Open Source](https://opensource.guide/how-to-contribute/). If you are new to contributing to Axelar, we encourage you to read [`CONTRIBUTING.md`](./CONTRIBUTING.md) in the Axelar examples repository.
+
+
+Before you start contributing, you need to:
+
+1. Fork the repository and clone it to your local machine.
+2. Create a new branch for your contribution.
+3. Install the dependencies for the project as indicated in the [README.md](https://github.com/axelarnetwork/axelar-examples#readme) file.
+4. Make your changes and commit them to your local repository.
```
Comment (StephenFluin) [inline CONTRIBUTING.md:28]: Shouldn't you test before you commit?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+Once you've completed these steps, you can contribute to Axelar Examples!
+
+### Issues
+
+If you find a bug in the source code or a mistake in the documentation, you can help us by submitting an issue to our [GitHub Repository](https://github.com/axelarnetwork/axelar-example/issues).
```
Comment (StephenFluin) [inline CONTRIBUTING.md:34]: just link "submit an issue" and get rid of "github repository".

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+Once you've completed these steps, you can contribute to Axelar Examples!
+
+### Issues
+
+If you find a bug in the source code or a mistake in the documentation, you can help us by submitting an issue to our [GitHub Repository](https://github.com/axelarnetwork/axelar-example/issues). 
+
+When you open a new issue, please provide as much detail as possible, including steps to reproduce the problem and the expected behaviour. Even better, you can submit a Pull Request with a fix.
+
+### Pull Requests
```
Comment (StephenFluin) [inline CONTRIBUTING.md:38]: Put this before the issue section, because there's two types  of contributions, code contributions (which involves making a change and sharing it with us), and non-code contributions such as issues. We should group them at least in the ordering, I don't think they need their own headers or anything like that.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+Before you submit your Pull Request (PR) consider the following guidelines:
+
+1. Search [GitHub project](https://github.com/axelarnetwork/axelar-example/issues) for an open or closed PR related to your submission. You don't want to duplicate effort.
+2. Make your changes in a new git branch:
+   ```shell
+   git checkout -b my-fix-branch
+   ```
+3. Create your patch, **including appropriate test cases**.
```
Comment (StephenFluin) [inline CONTRIBUTING.md:47]: We should specify where we want those tests, and what types of tests we want, as only some code paths are tested today.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+4. Run the entire test suite, and ensure all tests pass.
```
Comment (StephenFluin) [inline CONTRIBUTING.md:48]: how?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+5. Commit your changes using a descriptive commit message that follows our [commit message conventions](https://www.conventionalcommits.org/en/v1.0.0/).
```
Comment (StephenFluin) [inline CONTRIBUTING.md:49]: I feel like this is too much detail about how to use git? Should we link to another guide?

Comment (StephenFluin) [inline CONTRIBUTING.md:49]: According to this spec (partially based on Angular! :) ), YOUR commit message should have been `docs: add contribution guideline`

docs is the types, as we don't really have scopes in this repo. Tense should be simplified "add" vs "added"

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+6. Push your branch to GitHub:
+   ```shell
+   git push origin my-fix-branch
+   ```
+7. If we suggest changes then:
+   * Make the required updates.
+   * Re-run the test suites to ensure tests are still passing.
+   * Rebase your branch and force push to your GitHub repository (this will update your Pull Request)
+   * Wait for the project maintainer to review your changes and merge your Pull Request.
+8. Your commit message should follow the pattern: `<type>[optional scope]`: `<description>`.
```
Comment (StephenFluin) [inline CONTRIBUTING.md:63]: Just link instead of specifying here.

Comment (StephenFluin) [reply CONTRIBUTING.md:63]: This doesn't match the specification you list above.

File: package-lock.json [hunk 10]
```diff
@@ -6547,6 +6547,7 @@
             "version": "4.0.5",
             "resolved": "https://registry.npmjs.org/bufferutil/-/bufferutil-4.0.5.tgz",
             "integrity": "sha512-HTm14iMQKK2FjFLRTM5lAVcyaUzOnqbPtesFIvREgXpJHdQm8bWS+GkQgIkfaBYRHuCnea7w8UVNfwiAQhlr9A==",
+            "hasInstallScript": true,
```
Comment (StephenFluin) [inline package-lock.json:6550]: Why is this change in this PR?

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+### Introduction
+
+`Axelar Examples` is a collection of examples and tutorials for the Axelar Network. Before you start, we encourage you to read the rest of this document, which contains information on how and what you can contribute.
+
+### Getting Started
+
+If you are new to the Axelar Network, we encourage you to read the [Axelar Documentation](https://docs.axelar.network/) to learn more about the Axelar Network and how to get started. 
+
+If you are new to contributing to open-source projects, we encourage you to read [How to Contribute to Open Source](https://opensource.guide/how-to-contribute/). If you are new to contributing to Axelar, we encourage you to read [`CONTRIBUTING.md`](./CONTRIBUTING.md) in the Axelar examples repository.
```
[the remark this answers is not in the document]
Comment (Olanetsoft) [reply CONTRIBUTING.md:20]: Updated.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Olanetsoft) [reply CONTRIBUTING.md:26]: Yes, they won't have permission to push to the org repository. That's why I suggested a fork.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Olanetsoft) [reply CONTRIBUTING.md:30]: Its. Updated.

File: CONTRIBUTING.md (hunk 3 above)
Comment (Olanetsoft) [reply CONTRIBUTING.md:28]: Step included. Updated

File: CONTRIBUTING.md (hunk 4 above)
Comment (Olanetsoft) [reply CONTRIBUTING.md:34]: Updated.

File: CONTRIBUTING.md (hunk 5 above)
Comment (Olanetsoft) [reply CONTRIBUTING.md:38]: Updated

File: CONTRIBUTING.md (hunk 6 above)
Comment (Olanetsoft) [reply CONTRIBUTING.md:47]: I need to remove this until we have a proper test setup on the project to avoid confusion. Manual test for now.

File: CONTRIBUTING.md (hunk 7 above)
Comment (Olanetsoft) [reply CONTRIBUTING.md:48]: I need to remove this until we have a proper test setup on the project to avoid confusion. Manual test for now.

File: CONTRIBUTING.md (hunk 8 above)
Comment (Olanetsoft) [reply CONTRIBUTING.md:49]: Noted.

File: CONTRIBUTING.md (hunk 9 above)
Comment (Olanetsoft) [reply CONTRIBUTING.md:63]: Got it. Updated

File: package-lock.json (hunk 10 above)
Comment (Olanetsoft) [reply package-lock.json:6550]: Not sure how that was added; it must have been when I did run `npm ci` or something. Will revert.

**Review by StephenFluin — APPROVED** [review]

**Review by StephenFluin — APPROVED** [review]
Can you remove the package-lock.json modifications from this PR?

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+### Table of Contents
+  * [Introduction](#introduction)
+  * [Getting Started](#getting-started)
+  * [How to Contribute](#how-to-contribute)
+  * [Code of Conduct](#code-of-conduct)
+
+### Introduction
+
+`Axelar Examples` is a collection of examples and tutorials for the Axelar Network. Before you start, we encourage you to read the rest of this document, which contains information on how and what you can contribute.
```
Comment (StephenFluin) [inline CONTRIBUTING.md:13]: Does it actually include any tutorials? I think it's just examples.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+   ```shell
+   git checkout -b my-fix-branch
+   ```
+4. Make your changes in the new git branch 
+5. Commit your changes. Your commit message should follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
+
+   ```shell
+   git commit -a
+   ```
+Note: the optional commit `-a' command line option will automatically "add" and "rm" edited files.
```
Comment (StephenFluin) [inline CONTRIBUTING.md:37]: mismatch between backtick (`) and single quote (')

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+
+Before you submit your Pull Request (PR) consider the following guidelines:
+
+1. Search the [project](https://github.com/axelarnetwork/axelar-examples/pulls) for an open or closed PR related to your submission. You don't want to duplicate effort.
```
Comment (StephenFluin) [inline CONTRIBUTING.md:41]: This step should go BEFORE you start working

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+2. Push your branch to GitHub with the following command: 
+   ```shell
+   git push origin my-fix-branch
+   ```
+Open a pull request from your forked repository to the original repository.
+
+If we suggest changes then:
+   * Make the required updates and commit them to your branch.
+   * Wait for the project maintainer to review your changes and merge your Pull Request.
```
Comment (StephenFluin) [inline CONTRIBUTING.md:50]: "You may need to rebase your PR on the latest `main`."

File: package-lock.json [hunk 16]
```diff
@@ -21326,4 +21326,4 @@
             "dev": true
         }
     }
-}
+}
```
Comment (StephenFluin) [inline package-lock.json:21329]: Can you remove this unrelated change from your PR?

**Review by StephenFluin — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 13 above)
Comment (Olanetsoft) [reply CONTRIBUTING.md]: Updated

File: CONTRIBUTING.md (hunk 15 above)
Comment (StephenFluin) [reply CONTRIBUTING.md:47]: @Olanetsoft What about this comment for the list of actions?
